src/iteration_summary.jl

```diff
@@ -2,8 +2,9 @@ using ..Parameters: transform_to_unconstrained
 
 using Oceananigans.Utils: prettysummary
 
-struct IterationSummary{U, P, M, C, V, E, O}
+struct IterationSummary{U, G, P, M, C, V, E, O}
     parameters_unconstrained :: U
+    forward_map_output :: G
```

Comment:
Are we worried this could consume a lot of memory for large outputs / high iteration counts? (Maybe it won't matter if the iteration count is relatively small.) That was our previous reasoning for not adding it here.

Reply:
Yes, good point. Users (aka me) are going to want this kind of convenience if they want to emulate the forward map output. Maybe we could provide a wrapper that runs `iterate!` and stores the forward map output at each iteration so that it doesn't have to be automatically stored in the iteration summaries. Or maybe `forward_map_output` should default to `nothing` unless we set some keyword argument `save_forward_map_output` to `true`.

Reply:
I think the simplest option is what you suggested: implement a flag that says "don't save it". That protects users who might, potentially, have issues.

In the longer run, it seems like we actually need something akin to Oceananigans.Simulation with the ability to write / save output and execute callbacks during `iterate!`.

Another possibility is to eliminate `iterate!` and instead implement something like `pseudo_step!`. Then, the onus is on users to write things like

```Julia
for step = 1:10
    pseudo_step!(eki)
    # any analysis needed
end
```

This interface is similar to what dedalus provides --- it encourages users to "write their own callbacks".